### ISSUE TITLE
Add new `requirements` section, deprecate `runtime`

### ISSUE THREAD

**Issue by jdidion** [openwdl/wdl#540]
In the 3/10/23 governance call we decided that, rather than backport the changes to `runtime` that currently appear in the development version, we will instead deprecate the `runtime` section and add a new `requirements` section that will have the semantics of the development `runtime` section.

**Comment by DavyCats** [comment]
Changing from `runtime` to `requirements` would constitute a breaking change, so this would have to be part of 2.0, right? Unless `requirements` gets added as a synonym in 1.2 while `runtime` remains available as well.

**Comment by jdidion** [comment]
Yes, sorry, should have added more details. runtime will be deprecated but still supported in 1.2. It will be removed in 2.0.

**Comment by patmagee** [comment]
@DavyCats as @jdidion mentioned, we are not going to be breaking the spec here but will introduce an explicit deprecation of the `runtime` block in favour of the `requirements` section. 

They will co-exist up until the 2.0 release. `requirements` provide a stricter vocabulary hopefully helping with portability

**Comment by vsmalladi** [comment]
Im not sure this is necessary. This seems unnecessary to move from `runtime` to `requirements` what is the value we are getting by doing this?

**Comment by jdidion** [comment]
Originally, we had planned in WDL 2.0 to change the semantics of `runtime` (in a breaking way) to only allow specific attributes. All optional attributes were to move to `hints`.

We then decided to implement this change in WDL 1.2, but since we cannot introduce a breaking change, we decided instead to deprecate `runtime` in favor of `requirements` and `hints`. The name `requirements` is also more clear than `runtime` regarding the intent of that section, and it also matches the term used in `CWL`.
